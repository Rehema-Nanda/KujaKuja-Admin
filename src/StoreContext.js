// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
