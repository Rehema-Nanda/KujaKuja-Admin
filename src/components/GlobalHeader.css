.global-header {
    background-color: #ffc300;
    z-index: 1049;
}

.global-header.z-index-lower{
    z-index: 1047;
}

.navbar {
    min-height: 104px;
    border-top-width: 6px;
    border-top-style: solid;
}

.logo-parent-div{
    float: left;
}

.customer-logo-parent-div{
    float: left;
    height: 68px;
    width: 80px;
    margin-left: -16px;
    padding: 12px 12px 11px 12px;
    border-radius: 0px 8px 8px 0px;
    text-align: center;
}

.customer-logo {
    max-width: 100px;
    max-height: 100%;
    margin: auto;
}

.custom-logo{
    position: relative;
    top: 50%;
    transform: translateY(-50%);
}

.customer-logo-nav-link {
    padding: 0;
    display: inline-block;
    height: 100%;
}

.kuja-kuja-logo-nav-link {
    padding: 12px 0 0 12px;
    float: left;
}

.customer-logo-nav-link span.title-text {
    font-family: 'Nunito Sans', sans-serif !important;
    padding-left: 10px;
    color: #000000;
    font-size: 16px;
}

.navbar-toggler {
    padding: 0;
    margin-right: 16px;
    border: none;
    outline: none;
}

.global-header .navbar-toggler-icon {
    width: 32px;
    margin-right: 0;
    background-image: url('../img/hamburger.svg');
    background-repeat: no-repeat;
    background-size: 100% auto;
}

nav .navbar-nav li a {
    font-family: 'Nunito Sans', sans-serif;
    color: #171D21 !important;
    font-size: 16px;
    margin-right: 16px;
}

nav .navbar-nav li a.navbar_active {
    border-bottom: #171D21 1px solid;
}

nav .collapse.show .navbar-nav {
    margin-top: 32px;
    margin-bottom: 32px;
}

nav .collapse.show .navbar-nav li {
    padding-left: 24px;
}

nav .collapse.show .navbar-nav li a {
    margin: 16px;
    padding: 0;
}

.dropdown-menu{
    min-width: 192px;
    padding: 8px;
    border: none;
    border-radius: 8px;
    box-shadow: 0 18px 24px -6px rgba(23,29,33,0.14);
    z-index: 999999;
}

.dropdown-item {
    text-align: center;
    padding: 8px;
}

.dropdown-item:active {
    background-color: #F8F9FA;
}

.modal-body {
    padding: 16px !important;
}
