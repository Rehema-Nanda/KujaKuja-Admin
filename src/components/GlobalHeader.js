import React from "react";
import { NavLink as Link } from "react-router-dom";
import {
    Button, Collapse, DropdownItem, DropdownMenu, DropdownToggle, Modal,
    ModalBody, ModalFooter, Nav, Navbar, NavbarToggler, NavItem, NavLink, UncontrolledDropdown,
} from "reactstrap";
import { observer } from "mobx-react";
import "./GlobalHeader.css";
import reactCSS from "reactcss";
import AppConfig from "../AppConfig";
import { StoreContext } from "../StoreContext";

const kujakujaLogo = require("../img/global-logo-dark.svg");

class GlobalHeader extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            isMenuOpen: false,
            isLogoutConfirmationModalOpen: false,
        };

        this.toggleMenu = this.toggleMenu.bind(this);
        this.toggleLogoutConfirmationModal = this.toggleLogoutConfirmationModal.bind(this);
        this.handleLogout = this.handleLogout.bind(this);
    }

    toggleMenu() {
        const { isMenuOpen } = this.state;

        // close menu if mobile
        if (window.innerWidth <= 992) {
            this.setState({
                isMenuOpen: !isMenuOpen,
            });
        }
    }

    toggleLogoutConfirmationModal() {
        const { isLogoutConfirmationModalOpen } = this.state;

        this.setState({
            isLogoutConfirmationModalOpen: !isLogoutConfirmationModalOpen,
        });
    }

    async handleLogout() {
        const { apiService } = this.context;

        this.toggleLogoutConfirmationModal();
        await apiService.logout();
    }

    render() {
        const { userStore: { userInfo, apiService: { isAuthenticated } }, configStore: { siteHeader } } = this.context;
        const {
            isMenuOpen, isLogoutConfirmationModalOpen,
        } = this.state;

        const styles = reactCSS({
            default: {
                customHeaderBg: {
                    borderTopColor: siteHeader.highlightColour || AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                },
                customLogoParentBg: {
                    backgroundColor: siteHeader.highlightColour || AppConfig.SITE_HEADER_DEFAULT_HIGHLIGHT_COLOUR,
                },
            },
        });

        return (

            <div>
                <Navbar className="global-header" style={styles.customHeaderBg} expand="lg">
                    <div className="logo-parent-div">
                        {siteHeader.logoUrl
                        && (
                            <div style={styles.customLogoParentBg} className="customer-logo-parent-div">
                                <NavLink className="customer-logo-nav-link" to="/" tag={Link} onClick={this.toggleMenu}>
                                    <img alt="Logo" className="customer-logo custom-logo" src={siteHeader.logoUrl} />
                                </NavLink>
                            </div>
                        )}
                        <NavLink className="kuja-kuja-logo-nav-link" to="/" tag={Link} onClick={this.toggleMenu}>
                            <img alt="Logo" className="customer-logo" src={kujakujaLogo} />
                        </NavLink>
                    </div>

                    {isAuthenticated()
                    && (
                        <>
                            <NavbarToggler onClick={this.toggleMenu} />

                            <Collapse isOpen={isMenuOpen} navbar>
                                <Nav className="ml-auto" navbar>
                                    <NavItem>
                                        <NavLink
                                            to="/mydata"
                                            tag={Link}
                                            activeClassName="navbar_active"
                                            onClick={this.toggleMenu}
                                        >
                                            My Data
                                        </NavLink>
                                    </NavItem>
                                    <UncontrolledDropdown nav inNavbar>
                                        <DropdownToggle nav caret>
                                            Administration
                                        </DropdownToggle>
                                        <DropdownMenu>
                                            <DropdownItem
                                                href="/users"
                                                to="/users"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Team Members
                                            </DropdownItem>
                                            <DropdownItem
                                                href="/featured_ideas"
                                                to="/featured_ideas"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Featured Ideas
                                            </DropdownItem>
                                            <DropdownItem
                                                href="/countries"
                                                to="/countries"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Countries
                                            </DropdownItem>
                                            <DropdownItem
                                                href="/locations"
                                                to="/locations"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Locations
                                            </DropdownItem>
                                            <DropdownItem
                                                href="/service_points"
                                                to="/service_points"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Service Points
                                            </DropdownItem>
                                            <DropdownItem
                                                href="/service_types"
                                                to="/service_types"
                                                tag={Link}
                                                activeClassName="navbar_active"
                                                onClick={this.toggleMenu}
                                            >
                                                Service Types
                                            </DropdownItem>

                                            {userInfo.length > 0 && userInfo[0].isAdmin
                                            && (
                                                <>
                                                    <DropdownItem
                                                        href="/config"
                                                        to="/config"
                                                        tag={Link}
                                                        activeClassName="navbar_active"
                                                        onClick={this.toggleMenu}
                                                    >
                                                        Custom Header
                                                    </DropdownItem>
                                                    <DropdownItem
                                                        href="/data_review"
                                                        to="/data_review"
                                                        tag={Link}
                                                        activeClassName="navbar_active"
                                                        onClick={this.toggleMenu}
                                                    >
                                                        Data Review
                                                    </DropdownItem>
                                                    <DropdownItem
                                                        href="/tag_filters"
                                                        to="/tag_filters"
                                                        tag={Link}
                                                        activeClassName="navbar_active"
                                                        onClick={this.toggleMenu}
                                                    >
                                                        Tag Filter
                                                    </DropdownItem>
                                                </>
                                            )}
                                        </DropdownMenu>
                                    </UncontrolledDropdown>
                                    <NavItem>
                                        <NavLink
                                            to="#"
                                            tag={Link}
                                            onClick={this.toggleLogoutConfirmationModal}
                                        >
                                            Log Out
                                        </NavLink>
                                    </NavItem>
                                </Nav>
                            </Collapse>
                        </>
                    )}
                </Navbar>

                <Modal isOpen={isLogoutConfirmationModalOpen} toggle={this.toggleLogoutConfirmationModal}>
                    <ModalBody>
                        Do you really want to log out?
                    </ModalBody>
                    <ModalFooter>
                        <Button color="primary" className="primary" onClick={this.handleLogout}>Yes</Button>
                        <Button color="primary" className="primary cancel" onClick={this.toggleLogoutConfirmationModal}>No</Button>
                    </ModalFooter>
                </Modal>
            </div>

        );
    }
}

GlobalHeader.contextType = StoreContext;

export default observer(GlobalHeader);
