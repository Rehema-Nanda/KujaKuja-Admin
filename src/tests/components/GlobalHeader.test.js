import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import GlobalHeader from "../../components/GlobalHeader";
import { StoreProvider } from "../../StoreContext";

describe("<GlobalHeader />", () => {
    function WrapwithProviders({ children }) {
        return (
            <StoreProvider>
                <MemoryRouter>
                    {children}
                </MemoryRouter>
            </StoreProvider>
        );
    }
    const wrapper = mount(<GlobalHeader />, { wrappingComponent: WrapwithProviders });

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it.skip("should toggle isOpen state when navbar toggler is clicked", () => {
        wrapper.instance().isUserLoggedInNoRedirect = jest.fn(() => {
            return true;
        });
        wrapper.instance().forceUpdate();

        global.innerWidth = 500; // mock window innerwidth

        const { isOpen } = wrapper.state();
        wrapper.find("NavbarToggler").simulate("click");
        expect(wrapper.state().isOpen).toEqual(!isOpen);
    });

    it("should toggle modal state when logout toggle function is called", () => {
        const { isLogoutConfirmationModalOpen } = wrapper.state();
        const toggleLogoutConfirmationModal = jest.spyOn(wrapper.instance(), "toggleLogoutConfirmationModal");
        toggleLogoutConfirmationModal();
        expect(wrapper.state().isLogoutConfirmationModalOpen).toEqual(!isLogoutConfirmationModalOpen);
    });
});
