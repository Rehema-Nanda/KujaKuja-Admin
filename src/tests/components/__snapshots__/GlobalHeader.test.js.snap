// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalHeader /> renders correctly 1`] = `
"<GlobalHeader>
  <div>
    <Navbar className=\\"global-header\\" style={{...}} expand=\\"lg\\" tag=\\"nav\\">
      <nav style={{...}} className=\\"global-header navbar navbar-expand-lg\\">
        <div className=\\"logo-parent-div\\">
          <NavLink className=\\"kuja-kuja-logo-nav-link\\" to=\\"/\\" tag={[Function: NavLink]} onClick={[Function: bound toggleMenu]}>
            <NavLink to=\\"/\\" onClick={[Function: bound onClick]} className=\\"kuja-kuja-logo-nav-link nav-link\\" activeClassName=\\"active\\" aria-current=\\"page\\">
              <Route path=\\"\\\\\\\\/\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                <Link to=\\"/\\" className=\\"kuja-kuja-logo-nav-link nav-link active\\" style={{...}} aria-current=\\"page\\" onClick={[Function: bound onClick]} replace={false}>
                  <a className=\\"kuja-kuja-logo-nav-link nav-link active\\" style={{...}} aria-current=\\"page\\" onClick={[Function]} href=\\"/\\">
                    <img alt=\\"Logo\\" className=\\"customer-logo\\" src={{...}} />
                  </a>
                </Link>
              </Route>
            </NavLink>
          </NavLink>
        </div>
      </nav>
    </Navbar>
    <Modal isOpen={false} toggle={[Function: bound toggleLogoutConfirmationModal]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
  </div>
</GlobalHeader>"
`;
